modules/mobile/spec/support/helpers/rails_helper.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative 'spec_helper'
+require_relative 'sis_session_helper'
+require_relative '../../support/matchers/json_schema_matcher'
+
+RSpec.configure do |config|
+  config.define_derived_metadata(file_path: %r{modules/mobile/spec}) do |metadata|
+    metadata[:mobile_spec] = true
+  end
+
+  # Many specs have been known to inconsistently fail without this flipper enabled.
+  # Not every spec needs it but no specs need it disabled so to ensure we don't get flaky specs
+  # this will just be enabled for all specs
+  config.before :each, mobile_spec: true, type: :request do
+    Flipper.enable('va_online_scheduling')
+  end
+
+  config.after :each, mobile_spec: true, type: :request do |example|
+    content_type = response.header['Content-Type']
+
+    if content_type != 'application/pdf' && response.body.present? &&
```

Comment:
I think `content_type != 'application/pdf'` makes it a little more obvious exactly why that condition is necessary. The only two content types in our api is json and pdf so functionally I don't think it'll do anything. I've got a ticket, https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8530 to look into these conditions to see if they are truely needed or a fault of json api validation gem and want to make sure it's obvious to whoever picks it up that pdfs are the cause of a failure.

Reply:
I don't agree that this is the more obvious way. Imagine that we add a new endpoint that allows a user to download their service records as text files with a `text/plain` header (or whatever). We would have to change this to make it work. The functionality is specific to json, so I think it would be better to guard the behavior based upon it having a compatible header rather than assume that we'll never have a different header type. 

I hear what you're saying about there being some remaining ambiguity about whether this should work with a pdf or not. I haven't looked into the matter, but my guess is that we don't really expect pdf downloads to be jsonapi compliant and this functionality will continue to only apply when the content type is json.

Reply:
That seems like a desirable outcome to me. It forces the dev to evaluate if the new endpoint they're making should be nested in json or be a straight file and either consciously skip the validation or conform to it. And if in the ticket linked we find that all files responses can just be the file then it can be changed there to only validate on json responses.